psydac/api/discretization.py

```diff
@@ -177,9 +182,9 @@ def reduce_space_degrees(V, Vh, basis='B', sequence='DR'):
 
     if isinstance(V, VectorFunctionSpace):
         if isinstance(V.kind, (H1SpaceType, L2SpaceType, UndefinedSpaceType)):
-            Vh = ProductFemSpace(*[Vh]*V.ldim)
+            Wh = ProductFemSpace(*[Wh]*V.ldim)
```

Comment:
It appears that I should have left `Vh` on the right hand side here. But this seems confusing to me, because in this case we have not reduced the degree along any direction.

@saidctb Can you explain what is going on in this case?

Reply:
Actually, I've just realized that in the case of `L2SpaceType` the space `Vh` was reduced and hence `Wh` is correct here. In the case of `H1SpaceType` or `UndefinedSpaceType`, a reduced space `Wh` was not created and therefore we should use the original `Vh`.

Reply:
we use this in the case of a `VectorFunctionSpace` with the types `H1SpaceType, L2SpaceType, UndefinedSpaceType`, the spaces need to be repeated `V.ldim` times, your changes are correct, it just doesn't work when you have `H1SpaceType` because there is no reduction there, where `wh` == `vh`

Reply:
Perfect, I'll fix it now

Reply:
don't forget to include the `UndefinedSpaceType` also, because we don't do any reduction there either

Reply:
After commit c80d5af everything works again.